feature-libs/product/configurators/rulebased/cpq/cpq-configurator-serializer.ts

```diff
@@ -62,14 +63,28 @@ export class CpqConfiguratorSerializer
     return updateAttribute;
   }
 
+  processSelectedSingleValue(singleValue: string): string {
+    let processedValue: string = singleValue;
+    if (!processedValue) {
+      processedValue = VALUE_SEPARATOR;
+    }
+    return processedValue;
+  }
+
   prepareValueIds(attribute: Configurator.Attribute): string {
-    let valueIds = '';
+    let valueIds;
     const selectedValues: Configurator.Value[] = attribute.values.filter(
       (value) => value.selected
     );
-    selectedValues.forEach((value) => {
-      valueIds += value.valueCode + VALUE_SEPARATOR;
-    });
+
+    if (selectedValues && selectedValues.length > 0) {
+      valueIds = '';
+      selectedValues.forEach((value) => {
+        valueIds += value.valueCode + VALUE_SEPARATOR;
+      });
+    } else {
+      valueIds = VALUE_SEPARATOR;
+    }
```

Comment:
The else part is strange. just returning the Value Separator itself, when there is no value seems odd to me?
Would have expected the empty string, or just some special value.

Reply:
This is exactly the value which is expected from the CPQ update API to remove all / last values. Other values (e.g. empty string) leads to the "bad request".